packages/client/src/App.test.tsx

```diff
@@ -1,9 +1,13 @@
 import { render, screen } from '@testing-library/react';
-import { SignInPage } from './pages';
+import { RouterProvider, createMemoryRouter } from 'react-router-dom';
+import { routes } from './router';
 
-const appContent = 'SignInPage';
+const router = createMemoryRouter(routes, {
+  initialEntries: ['/'],
+  initialIndex: 0,
+});
 
-test('Example test', async () => {
-  render(<SignInPage />);
-  expect(screen.getByText(appContent)).toBeDefined();
+test('Signin loading page test', async () => {
+  render(<RouterProvider router={router} />);
+  expect(screen.getByText('SignInPage')).toBeDefined();
 });
```

Comment:
Использование screen.getByText не учитывает возможность асинхронного рендеринга компонентов. Для таких случаев лучше использовать screen.findByText, который возвращает промис и ожидает появления элемента в DOM.

```suggestion

test('Sign in loading page test', async () => {
  render(<RouterProvider router={router} />);

  const signInElement = await screen.findByText('SignInPage');
  expect(signInElement).toBeDefined();
});```

Reply:
Учтём на будущее, когда будет писать тесты более обширные